data-model/src/main/scala/za/co/absa/enceladus/model/conformanceRule/package.scala

```diff
@@ -121,6 +122,14 @@ package object conformanceRule {
     override def withUpdatedOrder(newOrder: Int): ConformanceRule = copy(order = newOrder)
   }
 
+  case class FillNullsConformanceRule(order: Int,
+                                      controlCheckpoint: Boolean,
+                                      outputColumn: String,
+                                      inputColumn: String,
+                                      value: String) extends ConformanceRule {
+    override def withUpdatedOrder(newOrder: Int): ConformanceRule = copy(order = newOrder)
```

Comment:
Probably not important, but I'll ask anyway:

Why do you strip the specific type here? The trait implementation would, of course, be satisfied if `FillNullsConformanceRule` was used as the return type and the object would not need to be cast if used further.

Reply:
If I understand correctly, means can only provide you with a string and we don't care about types until we get into conformance, where we cast it at that point of time.